packages/vaadin-grid/test/keyboard-navigation.test.js

```diff
@@ -1956,17 +1956,42 @@ describe('keyboard navigation', () => {
 
       const spy = sinon.spy();
 
-      grid.addEventListener('cell-focus', (e) => {
+      const eventHandler = (e) => {
         spy();
 
         const context = e.target.getEventContext(e);
         expect(context).to.deep.equal(expectedContext);
         done();
-      });
+      };
+
+      grid.addEventListener('cell-focus', eventHandler);
+
+      left();
+
+      expect(spy.calledOnce).to.be.true;
+      grid.removeEventListener('cell-focus', eventHandler);
+    });
+
+    it('should contain event context in event detail', (done) => {
+      tabToBody();
+      right();
+
+      const spy = sinon.spy();
+
+      const eventHandler = (e) => {
+        spy();
+
+        expect(e.detail).not.to.be.empty;
```

Comment:
This is in general too vague for a check, it would also pass if detail is just an empty object, or a boolean. I would at least check that `detail.context` is defined, or is an object. We can also go one step further and check the full context object as is done in the test above.

Also the setup with running expectations in the event handler and then using a spy to make sure the event handler did run at all is a bit unintuitive. As an alternative we could use a variable in the closure to capture the event data from the event handler, and then just assert at the end of the test function. Something like:
```
let eventDetail;

const eventHandler = (e) => {
  eventDetail = e.detail;
}

expect(eventDetail).to.be.defined();
expect(eventDetail.context).to...
```

Reply:
It is also possible to use `spy.args` to check spy's arguments:

> **spy.args**
> Array of arguments received, spy.args[0] is an array of arguments received in the first call.

https://sinonjs.org/releases/latest/spies/

An example of how it would look like for your case:

```js
it('should contain event context in event detail', () => {
  const spy = sinon.spy();
  grid.addEventListener('cell-focus', spy);

  tabToBody();
  right();
  left();

  expect(spy.calledOnce).to.be.true;
  expect(spy.args[0][0]).to.deep.equal({
    detail: {
      context: expectedContext
    }
  });
});
```

You can also find more examples of how to write tests for Web Components in [this presentation](https://docs.google.com/presentation/d/1-o-5p9N5mYUraNgC6eJaR63LySKli99Ba4g1ulMRBNU/edit#slide=id.gab5768d3d4_0_63)